bootstrap_cfn/fab_tasks.py

```diff
@@ -193,6 +193,94 @@ def swap_tags(tag1, tag2):
+
+    Sets all internet facing hostnames to resolve to given maintenance_ip
+    """
+    cfn_config = get_config()
+    r53_conn = get_connection(R53)
+
+    cached_zone_ids = {}
+    for elb in cfn_config.data['elb']:
+        if not elb['scheme'] == 'internet-facing':
+            continue
+
+        record = "{0}.{1}".format(elb['name'], elb['hosted_zone'])
+        zone_id = get_cached_zone_id(r53_conn, cached_zone_ids, elb['hosted_zone'])
+        r53_conn.update_dns_record(zone_id, record, 'A', maintenance_ip)
+
+
+@task
+def exit_maintenance():
+    """
+    Exit maintenance mode
+
+    Sets internet-facing elbs hostnames
+    back to the ELB DNS alias
+    """
+    r53_conn = get_connection(R53)
+    elb_conn = get_connection(ELB)
+
+    cfn_config = get_config()
+    stack_name = get_stack_name()
+
+    # get all elbs for this stack from AWS
+    stack_elbs = {}
+    [stack_elbs.update({
+                       getattr(x, 'logical_resource_id'): getattr(x, 'physical_resource_id')
+                       }) for x in elb_conn.cfn.get_stack_load_balancers(stack_name)]
+
+    # filter stack related load balancers (as opposed to all stack elbs in the account)
+    full_load_balancers = elb_conn.conn_elb.get_all_load_balancers(
+        load_balancer_names=stack_elbs.values())
+
+    cached_zone_ids = {}
+    # loop through elb config entries and change internet facing ones
+    for elb in cfn_config.data['elb']:
+        if not elb['scheme'] == 'internet-facing':
+            continue
+        record = "{0}.{1}".format(elb['name'], elb['hosted_zone'])
+        # obtain physical name from dict lookup, by converting elb name into safe name
+        # into logical name
+        phys_name = stack_elbs[get_elb_safe_name(elb['name'])]
+
+        dns_name = [x.dns_name for x in full_load_balancers if x.name == phys_name][0]
+        zone_id = get_cached_zone_id(r53_conn, cached_zone_ids, elb['hosted_zone'])
+
+        # For record_value provide list of params as needed by function set_alias
+        # http://boto.readthedocs.org/en/latest/ref/route53.html#boto.route53.record.Record.set_alias
+        record_value = [
+            # alias_hosted_zone_id
+            R53.AWS_ELB_ZONE_ID[env.aws_region],
+            # alias_dns_name
+            dns_name,
+            # alias_evaluate_target_health (True/False)
+            False
+        ]
+        r53_conn.update_dns_record(zone_id, record, 'A', record_value, is_alias=True)
+
+
+def get_cached_zone_id(r53_conn, zone_dict, zone_name):
+    '''
+    If we are looping through ELBs we may just have different hostnames
+    in same zone, so feel free to cache it (and drink a shot because I said 'cache')
+    '''
+    if zone_dict.get(zone_name, False):
+        return zone_dict[zone_name]
+    # not found, look it up, cache it up ..
+    zone_dict[zone_name] = r53_conn.get_hosted_zone_id(zone_name)
+    return zone_dict[zone_name]
+
+
+def get_elb_safe_name(elb_name):
```

Comment:
`get_` should not be used in a function name unless it is an accessor method. This would be better named `escape_elb_name` or `make_safe_elb_name` or similar.